### PR TITLE
16.0 Preview 4. Up and down key do not work for editor if the old completion is turned on

### PR DESCRIPTION
**Pull request by ivanbasov**
Fixes: https://github.com/dotnet/roslyn/issues/33744

### CODE REVIEW

File: src/EditorFeatures/Core/CommandHandlers/AbstractCompletionCommandHandler.cs [hunk 1]
```diff
@@ -299,8 +299,7 @@ public void ExecuteCommand(SelectAllCommandArgs args, Action nextHandler, Comman
 
         public VSCommanding.CommandState GetCommandState(EscapeKeyCommandArgs args, Func<VSCommanding.CommandState> nextHandler)
         {
-            AssertIsForeground();
-            return GetCommandStateWorker(args, nextHandler);
+            return nextHandler();
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/CommandHandlers/AbstractCompletionCommandHandler.cs]: When I look through all of the `GetCommandState`s in this file, it's not immediately obvious to me why these 3 should specially just defer (including when execute will actually take action). Can you please add comments to explain why these ones are special?

Comment (ivanbasov) [reply src/EditorFeatures/Core/CommandHandlers/AbstractCompletionCommandHandler.cs]: They originally were not supported in the command handler. The corresponding old completion controller does not support `GetCommandState(args, nextHandler)` with `UpKeyCommandArgs/DownKeyCommandArgs/EscapeKeyCommandArgs`.

The original code removed in #33196 immediately deferred them as well.

Comment (jasonmalinowski) [reply src/EditorFeatures/Core/CommandHandlers/AbstractCompletionCommandHandler.cs]: I guess I was confused by this too -- why can't all the ones just call nextHandler? I see the code in GetCommandStateWorker defers to the controller when there's a session open, but is that just going to return effectively what nextHandler will return anyways?

Comment (sharwell) [reply src/EditorFeatures/Core/CommandHandlers/AbstractCompletionCommandHandler.cs]: I would prefer changes to this code wait for a future PR. It takes a few hours to validate changes still have correct behavior.

**Pull request merged**